tests/test_env.py

```diff
@@ -659,7 +606,13 @@ def test_are_installed_dependency_cycle_on_extra(tmpdir, tmpvenv_active_inherit:
     )
 
     requirements = [Requirement.parse("pkg[optional-pkg]")]
-    tmpvenv_active_inherit.install_from_index(requirements=requirements, index_urls=[pip_index.url])
+    tmpvenv_active_inherit.install_for_config(
+        requirements=requirements,
+        config=PipConfig(
+            index_url=pip_index.url,
+            use_system_config=True,  # we need an upstream for some packages
```

Comment:
This one I think we should really try to avoid. As far as I can see it's not required at the moment.

Reply:
seems ok